tpm2/src/util/tpm2_kmyth_object.c

```diff
@@ -34,6 +40,17 @@ void tpm2_init_kmyth_object_sensitive(TPM2B_AUTH object_auth,
   // For data, the data buffer size cannot be zero - we must populate the
   // buffer with data to be sealed and set the size to its length in bytes.
   sensitiveArea->sensitive.data.size = object_dataSize;
+  // TODO: Calling memcpy with either source or dest addresses set to NULL
+  // is technically undefined behavior. However, object_data is expected
+if ( (object_dataSize == 0) || (object_data == NULL) )
+{
+   sensitiveArea->sensitive.data.size = 0;
+}
+else
+{
+    sensitiveArea->sensitive.data.size = object_dataSize;
+    memcpy(&sensitiveArea.sensitive.data.buffer, object_data, sensitiveArea.sensitive.data.size);
+}
   memcpy(&sensitiveArea->sensitive.data.buffer, object_data,
```

Comment:
Need to git rid of the memcpy() outside of the if/else. Also, I think the indent on the added code needs to be fixed and we should probably remove the TODO comment if we believe the issue is addressed by the change.

Reply:
Yep, removed the extra memcpy, the TODO comment, and fixed `sensitiveArea` field references.